crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -525,6 +528,13 @@ private void addDescription(ObjectNode result, AnnotatedElement element) {
         }
     }
 
+    private void addPreserveUnknownFields(ObjectNode result, AnnotatedElement element) {
```

Comment:
Since this annotation is either true or false, I wonder if it is worth mandating its presence on all classes, so that this doesn't fall out of sync in the future?

If so we could follow a pattern similar to what @tombentley suggested here: https://github.com/strimzi/strimzi-kafka-operator/pull/2820#discussion_r411257202 with calling error if the annotation is not supplied.

Reply:
I think it would be better to just assume that any `Map<String, *>` typed-property needed he fields preserving and only use the annotation where that default was incorrect. That way there's less visual clutter and no gotchas.

That said, can anyone think of a use case for having a Map typed property which _didn't_ need fields preserving? If not then I suggest we forego the annotation completely for the time being and add it only when it turns out we need it.

Reply:
I assume that should be applied also to lists and not just maps?

Reply:
I believe it is not needed for lists. Only maps as unknown keys are embedded in maps, but we can strongly type the elements of a list.We do still run into the issue if we have a list of maps, but this is something to be avoided.

Reply:
Ok, makes sense I guess since array is just another type.

Reply:
Exactly! :)